### ISSUE TITLE
Suggestion: default `swap-storage` to `false`

### ISSUE THREAD

**Issue by zaikunzhang** [jlumbroso/free-disk-space#12]
Hello @jlumbroso ,

First of all, thank you for this fantastic action. It is vital in the development of [my project PRIMA](https://github.com/libprima/prima).

[README](https://github.com/jlumbroso/free-disk-space#readme) mentions the following. 
```
        # all of these default to true, but feel free to set to
        # "false" if necessary for your workflow
        android: true
        dotnet: true
        haskell: true
        large-packages: true
        docker-images: true
        swap-storage: true
```

I suggest defaulting `swap-storage: true` to `false`. Otherwise, some memory-consuming actions may get the runner shut down due to memory starvation --- it is not uncommon for a space-intensive action to be memory-hungry as well. This did happen in [PRIMA](https://github.com/libprima/prima), and it took a couple of days to figure out that `swap-storage: true` was the reason. 

If  `swap-storage: true` is going to be retained, then careful documentation may be necessary to highlight the above-mentioned caveat.

Many thanks!

Best regards,
Zaikun

**Comment by StevePotter** [comment]
I agree.  I would also suggest making `large-packages` false by default.  The reason for that is it takes a long time, like 8 minutes.  That could be improved by combining the apt commands, but regardless I think off by default makes sense, or maybe just document that large-pages takes a long time.

**Comment by jlumbroso** [comment]
Hi @zaikunzhang — you were right, and I'm sorry it took three years to say so.

**`swap-storage` now defaults to `false`.**

The reasoning I was working from, which I'd never written down anywhere you could have found it, was roughly *ask forgiveness rather than permission*: Delete as much as possible by default, on the assumption that anyone reaching for this action is already out of disk, and that if something turns out to be missing you flip one input and move on.

**Your report is what showed me where that reasoning stops working.** It assumes the cost of over-deleting is cheap *because you can tell what happened*. Every other option, when it bites, produces an error naming the thing that's missing. Removing swap produces a job that dies under memory pressure — nothing about swap, nothing about this action. You wrote that it took a couple of days to track down, and that is the whole argument: a default shouldn't be able to break something as basic as running out of memory, least of all invisibly. And since a job short on disk is disproportionately likely to be short on memory too, the option most likely to hurt was the one most likely to be switched on.

So the fix is the one you asked for in the first place, plus the documentation you offered as a fallback:

- `swap-storage` defaults to `false`. If you want swap reclaimed, ask for it: `swap-storage: true`.
- There's now a FAQ on what each option's side effects actually are, and the OOM case is written up there in the terms you described it.
- The side-effects FAQ now documents this case and credits your report. The reason for the change is also recorded in [ADR-0007](https://github.com/jlumbroso/free-disk-space/blob/main/docs/adr/0007-swap-storage-defaults-to-false-the-policy-acquires-a-boundary.md).

Closing this as fixed. Thank you for the report, for the diagnosis, for proposing both a fix *and* a fallback, and for the patience — and thank you for PRIMA, which is a much better reason for this action to exist than the one I built it for! 😇
